File: LibraryAppFrontend/src/app/authorization/AuthorizationDTO.ts
import { Checkout } from "../functionality/models"

export interface LoginRequest {
    email: string
    password: string
}

export interface LoginResult {
    success: boolean
    token: string
    userId: string
    isAdmin:boolean
}

export interface RegisterRequest {
    username: string
    email: string
    password: string
}

export interface User {
    userId: string,
    username: string,
    email: string,
    checkouts: Checkout[]
}

File: LibraryAPI/DTO/AuthorizationDTO.cs
using System.ComponentModel.DataAnnotations;
using LibraryApp.API.Application.Validators;

namespace LibraryApp.API.DTO  {

    public record RegistrationRequest {

        [Required]
        public string username { get; set; }

        [EmailValidator]
        public string email { get; set; }

        [Required]
        public string password { get; set; }

    }

    public record LoginRequest {

        [EmailValidator]
        public string email { get; set; }

        [Required]
        public string password { get; set; }

    }

    public record LoginResult {

        public bool success { get; set; }

        public String message { get; set; }

        public String token { get; set; }

        public String userId { get; set; }

        public bool isAdmin { get; set; }
    }


}